.travis.yml

```diff
@@ -32,44 +32,47 @@ env:
   - S2N_LIBCRYPTO=openssl-1.1.0 OPENSSL_ia32cap="~0x200000200000000" BUILD_S2N=true TESTS=integration GCC6_REQUIRED=true
   - S2N_LIBCRYPTO=openssl-1.1.x-master BUILD_S2N=true TESTS=integration GCC6_REQUIRED=true
   - S2N_LIBCRYPTO=openssl-1.1.0 LATEST_CLANG=true TESTS=fuzz
-  - TESTS=sawHMAC SAW_HMAC_TEST=md5    SAW=true GCC6_REQUIRED=false
-  - TESTS=sawHMAC SAW_HMAC_TEST=sha1   SAW=true GCC6_REQUIRED=false
-  - TESTS=sawHMAC SAW_HMAC_TEST=sha224 SAW=true GCC6_REQUIRED=false
-  - TESTS=sawHMAC SAW_HMAC_TEST=sha256 SAW=true GCC6_REQUIRED=false
-  - TESTS=sawHMAC SAW_HMAC_TEST=sha384 SAW=true GCC6_REQUIRED=false
-  - TESTS=sawHMAC SAW_HMAC_TEST=sha512 SAW=true GCC6_REQUIRED=false
-  - TESTS=sawDRBG                      SAW=true
-  - TESTS=sawHMACFailure               SAW=true
+
 
 matrix:
   exclude:
   - os: osx
     env: TESTS=ctverif
   - os: osx
-    env: TESTS=sawHMAC SAW_HMAC_TEST=md5    SAW=true GCC6_REQUIRED=false
-  - os: osx
-    env: TESTS=sawHMAC SAW_HMAC_TEST=sha1   SAW=true GCC6_REQUIRED=false
-  - os: osx
-    env: TESTS=sawHMAC SAW_HMAC_TEST=sha224 SAW=true GCC6_REQUIRED=false
-  - os: osx
-    env: TESTS=sawHMAC SAW_HMAC_TEST=sha256 SAW=true GCC6_REQUIRED=false
-  - os: osx
-    env: TESTS=sawHMAC SAW_HMAC_TEST=sha384 SAW=true GCC6_REQUIRED=false
-  - os: osx
-    env: TESTS=sawHMAC SAW_HMAC_TEST=sha512 SAW=true GCC6_REQUIRED=false
-  - os: osx
-    env: S2N_LIBCRYPTO=openssl-1.1.0 LATEST_CLANG=true TESTS=fuzz
-  - os: osx
-    env: TESTS=sawDRBG                      SAW=true  GCC6_REQUIRED=false
-  - os: osx
-    env: TESTS=sawHMACFailure               SAW=true
-  - os: osx
     env: S2N_LIBCRYPTO=openssl-1.1.x-master BUILD_S2N=true TESTS=integration GCC6_REQUIRED=true
   #This exception is because the test isn't finished yet, remove to turn on DRBG Saw tests
   - env: TESTS=sawDRBG                      SAW=true
+  include:
+  - os : linux
+    env : TESTS=sawHMAC SAW_HMAC_TEST=md5    SAW=true GCC6_REQUIRED=false
+    addons: &sawaddons
+      apt:
+        packages:
+          - clang-3.8
+          - llvm-3.8
+  - os : linux
+    env : TESTS=sawHMAC SAW_HMAC_TEST=sha1   SAW=true GCC6_REQUIRED=false
+    addons : *sawaddons
+  - os : linux
+    env : TESTS=sawHMAC SAW_HMAC_TEST=sha224 SAW=true GCC6_REQUIRED=false
+    addons : *sawaddons
+  - os : linux
+    env : TESTS=sawHMAC SAW_HMAC_TEST=sha256 SAW=true GCC6_REQUIRED=false
+    addons : *sawaddons
+  - os : linux
+    env : TESTS=sawHMAC SAW_HMAC_TEST=sha384 SAW=true GCC6_REQUIRED=false
+    addons : *sawaddons
+  - os : linux
+    env : TESTS=sawHMAC SAW_HMAC_TEST=sha512 SAW=true GCC6_REQUIRED=false
+    addons : *sawaddons
+  - os : linux
+    env : TESTS=tls SAW=true GCC6_REQUIRED=false
+    addons : *sawaddons    
+  - os : linux
+    env : TESTS=sawHMACFailure SAW=true
+    addons : *sawaddons    
   allow_failures:
     - os: osx
-    - env: S2N_LIBCRYPTO=openssl-1.1.x-master BUILD_S2N=true TESTS=integration GCC6_REQUIRED=true
```

Comment:
I don't think this should have been removed from the allowed_failures list.

Reply:
Agreed and fixed. I got mixed up with the line in exclude on the merge. 

As an additional note, I've switched over all of our runs to including the exact configuration rather than putting them on the matrix and then excluding the configurations I don't want. This should work for that configuration as well, to save a few lines of Travis configuration.